### PR TITLE
refactor: replace if statements with switch in cardError code

### PR DESCRIPTION
**Pull request by aritro2002**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

replaced if statements with switch in cardError code.

## How did you test it?

By compiling code and also via rendering the SDK in chrome.

## Checklist

- [x] I ran `npm run re:build`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: src/Payment.res [hunk 1]
```diff
@@ -328,13 +328,17 @@ let make = (~paymentMode, ~integrateError, ~logger) => {
...
+      isCardValid,
+      isCardSupported->Option.getOr(true),
+      isCardValid->Option.getOr(true),
+      cardNumber->String.length == 0,
+    ) {
+    | (None, _, _, _) => ""
+    | (_, _, _, true) => ""
+    | (_, true, true, _) => ""
+    | (_, true, _, _) => localeString.inValidCardErrorText
+    | (_, _, _, _) =>
```
Comment (PritishBudhiraja) [inline src/Payment.res:341]: I feel the previous one is more readable and better. This is more confusing

Comment (Sanskar2001) [reply src/Payment.res:341]: isCardValid seems redundant here, can we remove that? Also try to simplify this further

**Review by Sanskar2001 — CHANGES_REQUESTED** [review]

File: src/Payment.res (hunk 1 above)
Comment (PritishBudhiraja) [reply src/Payment.res:341]: Hey @aritro2002 

Try this -

```
  React.useEffect(() => {
    let cardError = if isCardValid == None || cardNumber->String.length == 0 {
      ""
    } else {
      let isCardSupportedValue = isCardSupported->Option.getOr(true)
      let isCardValidValue = isCardValid->Option.getOr(false)

      if isCardSupportedValue && isCardValidValue {
        ""
      } else if isCardSupportedValue {
        localeString.inValidCardErrorText
      } else {
        let cardBrand = cardNumber->CardUtils.getCardBrand
        if cardBrand == "" {
          localeString.inValidCardErrorText
        } else {
          localeString.cardBrandConfiguredErrorText(cardBrand)
        }
      }
    }
    setCardError(_ => cardError)
    None
  }, [isCardValid, isCardSupported])
```

**Review by PritishBudhiraja — DISMISSED** [review]
LGTM :)

File: src/Payment.res [hunk 2]
```diff
@@ -330,14 +330,21 @@ let make = (~paymentMode, ~integrateError, ~logger) => {
   React.useEffect(() => {
```
Comment (Sanskar2001) [inline src/Payment.res:330]: @aritro2002 can you please try this 
```
  React.useEffect(() => {
    let cardError = switch (
      isCardSupported->Option.getOr(true),
      isCardValid->Option.getOr(true),
      cardNumber->String.length == 0,
    ) {
    | (_, _, true) => ""
    | (true, true, _) => ""
    | (true, _, _) => localeString.inValidCardErrorText
    | (_, _, _) => CardUtils.getCardBrandInvalidError(~cardNumber, ~localeString)
    }
    if isCardValid->Option.isSome {
      setCardError(_ => cardError)
    }
    None
  }, [isCardValid, isCardSupported])
```

In CardUtils.res please define the below function
```
let getCardBrandInvalidError = (
  ~cardNumber,
  ~localeString: OrcaPaymentPage.LocaleStringTypes.localeStrings,
) => {
  switch cardNumber->getCardBrand {
  | "" => localeString.enterValidCardNumberErrorText
  | cardBrandValue => localeString.cardBrandConfiguredErrorText(cardBrandValue)
  }
}
```

**Review by Sanskar2001 — CHANGES_REQUESTED** [review]

File: src/Payment.res (hunk 2 above)
Comment (Sanskar2001) [reply src/Payment.res:330]: Also, please test it for all the supported card brands for both valid and invalid cards.

**Review by Sanskar2001 — APPROVED** [review]

**Review by PritishBudhiraja — APPROVED** [review]

**Pull request merged**